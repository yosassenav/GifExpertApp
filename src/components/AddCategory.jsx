import { useState } from "react"

export const AddCategory = ({onNewCategories}) => {

    const [inputValue, setInputValue] = useState('');

    const onInputChange = ({target}) => {
        //console.log(event.target.value);
        setInputValue(target.value);
    }

    const handleSubmit = (evt) => {
        evt.preventDefault();
        if (inputValue.trim().length <= 1) return;

        // setCategories(categories => [inputValue, ...categories])
        onNewCategories(inputValue.trim());
        setInputValue('');
    };

    return(
    <form onSubmit={handleSubmit}>
        <input
        type="text"
        placeholder="Buscar Gifs"
        onChange={onInputChange}
        value={inputValue}
        />
        
    </form>
    )
}