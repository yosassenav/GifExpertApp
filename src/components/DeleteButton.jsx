import PropTypes from "prop-types";


export const DeleteButton = ({onClick}) => {
    return <button onClick={onClick} className="delete-button">Delete Category</button>
}

DeleteButton.propTypes = {
    onClick: PropTypes.func,
}