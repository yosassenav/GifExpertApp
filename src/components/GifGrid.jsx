import PropTypes from "prop-types";
import { getGifs } from "../helpers/getGifs"


export const Gifgrid = ({category}) => {

    getGifs(category);
    
    return (
        <>
            <h3>{category}</h3>
            <li>{category}</li>
        </>
    )
}

Gifgrid.propTypes = {
    category: PropTypes.string.isRequired,
  };