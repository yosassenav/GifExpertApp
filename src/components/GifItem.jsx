import PropTypes from "prop-types";

export const GifItem = ({title, url}) => {

    return  (
        <div className="card">
            <img src={url} alt={title}/>
            <p>{title}</p>
        </div>
    );
   
};

GifItem.prototypes = {
    id: PropTypes.number,
    title : PropTypes.string,
    url: PropTypes.string,
};