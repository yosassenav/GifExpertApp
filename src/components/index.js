export * from "../components/AddCategory";
export * from "../components/GifGrid";
export * from "../components/GifItem";
export * from "../components/DeleteButton";

