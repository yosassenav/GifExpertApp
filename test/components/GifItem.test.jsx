import { render } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Pruebas en <GifItem/>',()=>{
    const title = 'Sailor Moon';
    const url = 'https://test-url.com/sailor-moon.jpg';

    test('debe hacer match con el snapshot',()=>{
        const { container } = render(<GifItem title={title} url={url}/>);
        expect(container).toMatchSnapshot();

    })

});