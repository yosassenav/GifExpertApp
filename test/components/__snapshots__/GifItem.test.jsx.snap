// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem/> debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Sailor Moon"
      src="https://test-url.com/sailor-moon.jpg"
    />
    <p>
      Sailor Moon
    </p>
  </div>
</div>
`;
